sys/trace/trace.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2020 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Freie Universität Berlin
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_util
+ * @{
+ *
+ * @file
+ * @brief       Execution tracing module implementation
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "irq.h"
+#include "xtimer.h"
+
+#ifndef TRACE_BUFSIZE
+#define TRACE_BUFSIZE 512
+#endif
```

Comment:
Why not have this in the header?

Reply:
because it is not part of the interface. should it be?

Reply:
Don't know, so lets leave it that way for now.